### PR TITLE
JpaRepository used, Liquibase support added, soft delete added, searc…

### PR DESCRIPTION
**Pull request by validatorr**
…h and delete endpoints added

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -59,6 +59,16 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>${liquibase.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-maven-plugin</artifactId>
+            <version>${liquibase.version}</version>
```
Comment (pavlov-n) [inline pom.xml]: you haven't such property like liquibase.version

File: src/main/java/v/hryhoryk/onlinebookstore/repository/book/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package v.hryhoryk.onlinebookstore.repository.book;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import v.hryhoryk.onlinebookstore.model.Book;
+
```
Comment (pavlov-n) [inline src/main/java/v/hryhoryk/onlinebookstore/repository/book/BookRepository.java:7]: i think here should be annotation Repository

**Review by pavlov-n — APPROVED** [review]
Looks good

File: pom.xml (hunk 1 above)
Comment (liliia-ponomarenko) [reply pom.xml]: can be removed

File: src/main/java/v/hryhoryk/onlinebookstore/mapper/BookMapper.java [hunk 3]
```diff
@@ -14,5 +15,7 @@
 public interface BookMapper {
     BookDto toDto(Book book);
 
-    Book toModel(CreateBookRequestDto requestDto);
+    Book toEntity(CreateBookRequestDto requestDto);
```
Comment (liliia-ponomarenko) [inline src/main/java/v/hryhoryk/onlinebookstore/mapper/BookMapper.java]:
```suggestion
    Book toBook(CreateBookRequestDto requestDto);
```

File: src/main/java/v/hryhoryk/onlinebookstore/model/Book.java [hunk 4]
```diff
@@ -25,5 +33,10 @@ public class Book {
     @Column(nullable = false)
     private BigDecimal price;
     private String description;
+    @Column(name = "cover_image")
     private String coverImage;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
 }
+
+
```
Comment (liliia-ponomarenko) [inline src/main/java/v/hryhoryk/onlinebookstore/model/Book.java]: Redundant  empty lines

File: src/main/java/v/hryhoryk/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+import v.hryhoryk.onlinebookstore.model.Book;
+import v.hryhoryk.onlinebookstore.repository.SpecificationProvider;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/v/hryhoryk/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java]:
```suggestion
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
good job! Let’s  fix a few mistakes ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Great 😊

**Pull request merged**